File: js_sequelize/models/Student.js
const {database} = require("../database");
const {DataTypes} = require("sequelize");

module.exports.Student = database.define('Student', {
    user_ID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
    },
    field_of_study: {
        type: DataTypes.STRING(50),
        allowNull: false
    },
    semester: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    year_of_study: {
        type: DataTypes.STRING(10),
        allowNull: false
    },
    mode_of_study: {
        type: DataTypes.STRING(20),
        allowNull: false
    }
}, {
    tableName: 'student_info',
    timestamps: false
});

File: dotnet/entity-framework/Entities/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_entity_framework.Entities;

[Table("student_info")]
public class Student
{
    [Key, ForeignKey("User")]
    public int UserId { get; set; }
    public string FieldOfStudy { get; set; }
    public int Semester { get; set; }
    public string YearOfStudy { get; set; }
    public string ModeOfStudy { get; set; }

    public User User { get; set; } = null!;
}
